lib/collection/src/collection/mod.rs

```diff
@@ -699,6 +703,54 @@ impl Collection {
         Ok(())
     }
 
+    pub async fn abort_resharding(
+        &self,
+        peer_id: PeerId,
+        shard_id: ShardId,
+        shard_key: Option<ShardKey>,
+    ) -> CollectionResult<()> {
+        let mut shard_holder = self.shards_holder.write().await;
+
+        if let Some(state) = self.resharding_state.read().deref() {
+            let is_in_progress = state.peer_id == peer_id
+                && state.shard_id == shard_id
+                && state.shard_key == shard_key;
+
+            if !is_in_progress {
+                return Err(CollectionError::bad_request(format!(
+                    "resharding of collection {} is not in progress",
+                    self.id,
+                )));
+            }
+        } else {
+            log::warn!(
+                "aborting resharding of collection {} ({peer_id}/{shard_id}/{shard_key:?}),\
+                 but resharding is not in progress",
+                self.id,
+            );
+        };
+
+        if shard_holder.get_shard(&shard_id).is_none() {
+            log::warn!(
+                "aborting resharding of collection {} ({peer_id}/{shard_id}/{shard_key:?}), \
+                 but shard {shard_id} does not exist in collection",
+                self.id,
+            );
+        }
+
+        // TODO: Contextualize errors? 🤔
+        shard_holder
+            .abort_resharding(shard_id, peer_id, shard_key.clone())
+            .await?;
+
+        // TODO: Contextualize errors? 🤔
```

Comment:
```suggestion
```

Reply:
I'd leave this for now. Will add `TODO(resharding)` tag though.